### DISCUSSION TITLE
How can I use regexp to find the beginning and end of the containing the line feed?

### DISCUSSION THREAD

**Discussion by chl3188** [microsoft/monaco-editor#3836] [Q&A] [answered]
I configured the provisionCodeLenses as follows to use the codeelense function.

```
provideCodeLenses(model, token) {
          let list: CodeLens[] = [];
          const matches = model.findMatches(
            "^(SELECT|INSERT)[\\s\\S]*?;$",
            true,
            true,
            false,
            null,
            true
          );

          if (!matches) {
            return [];
          }

          matches.forEach((it, index) => {
            list.push({
              range: {
                startLineNumber: it.range.startLineNumber,
                startColumn: 1,
                endLineNumber: it.range.endLineNumber,
                endColumn: 1,
              },
              id: "lense-command-id-1",
              command: {
                id: "1",
                title: "SQL Execute",
              },
            });
          });

          return {
            lenses: list,
            dispose: () => {},
          };
        },
```
It's a regex that I made.

`/^(SELECT|INSERT)[\s\S]*?;$/gmi`

and this is sample text

```
SELECT *
FROM users
WHERE age > 18
AND gender = 'F';
```

I checked that the test site works normally with that regular expression.
https://regex101.com/
However, if apply it to findMatches in the actual editor, it doesn't work if there is an opening in the SQL sentence.
What's the problem?
pls Help me

**Comment by tamayika** [comment, marked as answer]
Please use this regex.
```
^(SELECT|INSERT)[\\n\\s\\S]*?;$
```
vscode optimises search algorithm by input regex pattern.
If `\r` or `\n` exits in pattern, vscode search multiline texts but otherwise search singleline.

https://github.com/microsoft/vscode/blob/13ba7bb446a638d37ebccb1a7d74e31c32bb9790/src/vs/editor/common/model/textModelSearch.ts#L69-L99.

**Reply by tamayika** [reply]
I thought this was a bug in vscode but `\s` is intentionally treated as not multiline character. (https://github.com/microsoft/vscode/blob/main/src/vs/editor/test/common/model/textModelSearch.test.ts#L742)

@roblourens (sorry for mentioning by blame)
Do you know why `\s` is considered as not multiline character?

**Reply by chl3188** [reply]
Wow, it's working. Thank you, sir !
